src/components/__tests__/ImageCarousel.spec.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { render } from '@testing-library/react-native'
+import ImageCarousel from '../ImageCarousel'
+
+describe('Components', () => {
+  const images = [
+    {
+      id: 1,
+      image: '1'
+    },
+    {
+      id: 2,
+      image: '2'
+    }
+  ]
+  describe('ImageCarousel', () => {
+    it('should render correctly', function () {
+      const { getAllByTestId } = render(<ImageCarousel images={images} />)
```

Comment:
Any other option than getByTestId for images?
Is it possible to trigger a swipe event?

Reply:
https://callstack.github.io/react-native-testing-library/docs/api-queries#bya11yrole-byaccessibilityrole-byrole should work I guess.

Reply:
unfortunately not.
Error: No instances found with accessibilityRole "image"

Reply:
Is using querySelector a possibility? https://medium.com/@drake_beth/how-to-test-images-in-react-a70053b1634a

Reply:
I don't know how to trigger a swipe event, would it be enough to just call the snapToNext method from snap-carousel?

Reply:
@hepperle00 I think that only works in the DOM and not on native. You can set `accessibilityRole='image'` on the images though and use `getByRole`.

Regarding the swipe events: AFAICT image carousel uses scroll views under the hood so you may be able to use simple `fireEvent.scroll`: https://callstack.github.io/react-native-testing-library/docs/api/#fireeventscroll-element-reacttestinstance-data-arrayany--void
Otherwise it seems to be possible to define custom events, but not sure if that works: https://callstack.github.io/react-native-testing-library/docs/api/#fireevent

Reply:
Just tested it, the following seems to work:
```
    const eventData = {
      nativeEvent: {
        contentOffset: {
          x: 200
        },
        contentSize: {
          // Dimensions of the scrollable content
          height: 500,
          width: 100
        },
        layoutMeasurement: {
          // Dimensions of the device
          height: 100,
          width: 100
        }
      }
    }
    fireEvent.scroll(getAllByRole('image')[0], eventData)

```

Reply:
yeah thanks. This works.